### DISCUSSION TITLE
Application freezes for ~10 seconds at startup

### DISCUSSION THREAD

**Discussion by makkus** [pthom/imgui_bundle#77] [Q&A] [answered]
Hiya, thanks for this library/framework, I really like the idea of a batteries included imgui distribution, that supports Python!

I've started working on a proof-of-concept UI for one of my projects, to test the waters, and everything went pretty well, and the development velocity is surprisingly high (having never used imgui before).

But I'm experiencing a weird issue, which I'm pretty sure is something in my setup (Fedora/Conda) and not a problem with imgui_bundle. But I'm at a loss where I could start debugging this, so I figured I'd give it a shot here. 

I don't know what I did, but at some stage my demo application started to take about 10 seconds to start up (before that it was in the order of 1 second, but that was mostly my own init stuff taking some time). Performance once fully started is crisp, but I can not explain the startup lag. I did a bit of debugging, tried to disable a lot of functionality of my app to figure out where the problem could be, couldn't find anything. On a hunch I tried the `demo_imgui_bundle` app again, and it turns out that this now also takes around 10 seconds to startup. It was much faster before.

Any idea where I could look, or how to pinpoint where/when the freeze happens? I re-installed my conda environment, rebooted my machine to rule out any weird networking issue or some-such, to no avail.

**Comment by pthom** [comment]
if also demo_imgu_bundle perf changed, something likely changed in your setup, and a 10 seconds lag makes me suspect a kernel lag (locked file, etc.) 

May be try diagnosing with strace and see if it block somewhere? Beware strace is really verbose; but sometimes useful

**Comment by makkus** [comment]
Ok, found the culprit. I'm not sure who or what put it there, but I had a `imgui.ini` file in my project directory. Contained lots of stuff like:

```
...
...
[Table][0x5DC92C7F,2]
RefScale=14
Column 0  Width=0
Column 1  Weight=1.0000

[Table][0x3B9BF61C,2]
RefScale=14
Column 0  Width=0
Column 1  Weight=1.0000

[Table][0xBAA35249,2]
RefScale=14
Column 0  Width=0
Column 1  Weight=1.0000

[Table][0xC0E68AB1,2]
RefScale=14
Column 0  Width=0
Column 1  Weight=1.0000

...
...
```

Once I deleted that file, all was good again. Any idea how that could have gotten there?

**Comment by pthom** [comment, marked as answer]
> I've started working on a proof-of-concept UI for one of my projects, to test the waters, and everything went pretty well, and the development velocity is surprisingly high (having never used imgui before).

Happy to read this!

> imgui.ini: Once I deleted that file, all was good again. Any idea how that could have gotten there?
imgui.ini is where imgui saves some of its settings (inner window position, etc).

You can change its name via `imgui.get_io().set_ini_filename`

Note: 
hello_imgui adds another file `imgui_app_window.ini` which stores the native app window position and size

**Reply by makkus** [reply]
Thanks, much appreciated! I'll have a look and see whether the ini file corruption shows up again, not sure what caused it to have thousands of entries in it. The current one seems and works fine.
